### PR TITLE
#60 in docker better rebased

### PR DESCRIPTION
**Pull request by xmik**
The discussion is in PR #148

### CODE REVIEW

File: libraries/helpers.rb [hunk 1]
```diff
@@ -81,14 +81,67 @@ def zap_extra_env_files
...
-    def wait_for_service
-      unless inside_docker?
-        sleep 1 until ::FileTest.pipe?("#{service_dir_name}/supervise/ok")
+    # Checks if runsvdir process is running. runsvdir
+    # starts and monitors services from #{parsed_service_dir}.
+    # (http://smarden.org/runit/runsvdir.8.html). 
+    # True e.g. in a docker container which uses Phusion 
+    # baseimage-docker, but false when running `docker build`. 
+    def runsvdir_running?
+      cmd = "ps -A -o command | grep \"^/usr/bin/runsvdir -P"\
```
Comment (coderanger) [inline libraries/helpers.rb]: This makes me deeply uncomfortable, at a minimum don't use grep.

Comment (xmik) [inline libraries/helpers.rb]: @coderanger I replaced the `ps ... | grep ...` command with verifying the `/proc/#{pid}/status` file. (See also the commit message).

**Comment by coderanger** [comment]
I think you went too far in the opposite direction. Using `ps` is okay, or at least better than implementing it yourself. My concern was you were parsing its output in Bash instead of in Ruby and generally not using the correct mixlib-shellout helpers (ex. `error?`).

**Comment by xmik** [comment]
@coderanger, I see. So do you want me to change this to using `ps` with mixlib-shellout but still avoid `grep`? Or do you need to apply more changes anyway?

**Comment by cwjohnston** [comment]
Closing in favor of #193

**Pull request closed without merging**